### ISSUE TITLE
Remove System.Interactive.Async dependency

### ISSUE THREAD

**Issue by vacamra** [okta/okta-sdk-abstractions-dotnet#17]
Hello,

TL;DR:
I would like to ask you to remove the System.Interactive.Async unused dependency from this library, as it causes ambiguity issues with Entity Framework Core when included in other projects (even transitively included). I believe this should not pose any issue, as the System.Interactive.Async library seems to not be used right now in the codebase.

Long version:
I have been trying to add Okta .NET SDK nuget package reference (Okta.Sdk v5.2.1) into my .NET Core ASP.Net + EFCore project. Upon adding the reference I have started receiving hundreds of compilations errors due to ambiguity between extension methods of System.Interactive.Async (who for some reason decided to add their code directly into the System.Linq namespace) and EntityFramework Core (who implement the same set of extension methods - in their own namespace). 

I am aware that the core of this issue issue should be solved by the creators of the System.Interactive.Async package, however they do not seem too keen on doing that, so we're left to work around those issues (e.g. by not using that package).

I have found out that Okta.Sdk references System.Interactive.Async only transitively via this Okta.Sdk.Abstractions project, which, while referencing the dependency directly, does not seem to actively use it.

Therefore I would like to ask you to remove this dependency so that importing Okta SDK won't break existing projects.

Thanks

**Comment by andriizhegurov-okta** [comment]
Hi @vacamra, thank you for the suggestion. I've put an item to our backlog to review the issue and take actions.
